mattermost.go

```diff
@@ -530,13 +545,19 @@ func convertTeamTo30(primaryTeamName string, team *TeamForUpgrade, uniqueEmails
 			}
 		}
 
+		if user.AuthData != nil && *user.AuthData != "" && uniqueAuths[*user.AuthData] {
+			shouldUpdateUser = true
+		}
+
 		if shouldUpdateUser {
 			if _, err := store.GetMaster().Exec(`
 				UPDATE Users 
 				SET 
 				    Email = :Email,
 				    Username = :Username,
-				    Roles = :Roles
+				    Roles = :Roles,
+				    AuthService = '',
```

Comment:
This is the case where we are changing the email or username and breaking the connection to LDAP or Gitlab.  In that case we blank the AuthData and AuthService fields to force it back to email login for duplicate accounts.